nupic/embodied/policies/auxiliary_tasks.py

```diff
@@ -0,0 +1,479 @@
+        self.param_list = []
+        if features_shared_with_policy:
+            # No need to nitialize a feature model because it is already initialized in
+            # curious_cnn_policy.py
+            self.features_model = None
+        else:
+            # Initialize small convolutional network for feature extraction from images
+            self.features_model = small_convnet(
+                self.ob_space,
+                nonlinear=torch.nn.LeakyReLU,
+                feat_dim=self.feat_dim,
+                last_nonlinear=None,
+                layernormalize=self.layernormalize,
+                device=self.device,
+            ).to(self.device)
+            # Add feature model to optimization parameters
+            self.param_list = self.param_list + [
+                dict(params=self.features_model.parameters())
+            ]
+
+        self.scope = scope
+
+        self.features = None
+        self.next_features = None
+        self.ac = None
+        self.ob = None
+
+    def update_features(self, obs, last_obs):
+        """Get features from feature model and set self.features and self.next_features.
+        Also sets self.ac to the last actions at end of rollout..
+
+        Parameters
+        ----------
+        obs : array
+            Current observations.
+        last_obs : array
+            Previous observations.
+
+        """
+        if self.features_shared_with_policy:
+            # Get features corresponding with the observation from the policy network
+            self.features = self.policy.flat_features
+            last_features = self.policy.get_features(last_obs)
+        else:
+            # Get features corresponding with the observation from the feature network
+            self.features = self.get_features(obs)
+            last_features = self.get_features(last_obs)
+        # concatenate the last and current features
+        self.next_features = torch.cat([self.features[:, 1:], last_features], 1)
+
+        self.ac = self.policy.ac
+        self.ob = self.policy.ob
+
+    def get_features(self, obs):
+        """Get features from the feature network.
+
+        Parameters
+        ----------
+        obs : array
+            Observation for which to get features.
+
+        Returns
+        -------
+        array
+            Features of the observations.
+
+        """
+        has_timesteps = len(obs.shape) == 5
+        if has_timesteps:
+            sh = obs.shape
+            obs = flatten_dims(obs, len(self.ob_space.shape))
+        # Normalize observations
+        obs = (obs - self.ob_mean) / self.ob_std
+        # Reshape observations
+        obs = np.transpose(obs, [i for i in range(len(obs.shape) - 3)] + [-1, -3, -2])
+        # Get features from the features_model
+        act = self.features_model(torch.tensor(obs).to(self.device))
+
+        if has_timesteps:
+            act = unflatten_first_dim(act, sh)
+
+        """ print(
```

Comment:
To make the print statements permanent, without having to comment, an option is to add a verbose flag and put the print statements under the flag.